### DISCUSSION TITLE
Play a youtube music playlist in automation

### DISCUSSION THREAD

**Discussion by hundredfire** [music-assistant/support#1886] [Q&A] [answered]
Home assistant supervised version: 2024.01.1
Music assistant BETA version:  2.0.0b92 

Hello everyone. I managed to install music assistant using a home assistant supervised container on my synology NAS. Everything is working well and I can play my youtube music playlists using the ingressed web interface of music assistant and cast it properly on my google nest. I even imported my youtube music playlists into music assistant's library and can find them in the respective category.

However, when I try to play a playlist in an automated task, I have a "Unkown media source" error.

I'm not sure I'm getting the syntax right in my yaml. I put this as media source and type:
`media_content_id: media-source://mass/database://playlist/4` where 4 is the number of the playlist in the order I added them into my library
`media_content_type: playlist`

I also tried putting the playlist name directly, as in : `media_content_id: playlist_name`. Then I can hear the chime of my Google nest connecting, but nothing is playing/nothing is queued. No "Unkown media source" message this time though

I'm guessing that I'm getting the syntax wrong somewhere, but I can't find what I'm doing wrong here.

edit: I should add that I tried to use both the service call and the appropriate "Play media" action (not home assistant's default Play media action)

**Comment by OzGav** [comment]
You can try the developer tools service call area to test this. Please show the full yaml for your service call if you need further assistance. You are correct you only need to use the playlist name

**Reply by hundredfire** [reply]
Thank you for answering so quickly. Here's my service call yaml:
 ```
service: media_player.play_media
target:
  device_id: media_player.nest_mini
data:
  media_content_id: Ballad
  media_content_type: playlist
```

Debug log doesn't show any error but only says:
` 2024-02-09 01:47:18.566 DEBUG (Thread-8) [music_assistant.providers.chromecast.Nest mini] Received media status update: IDLE
`
And numerous `DEBUG (MainThread) [music_assistant.event] media_item_added library://track/[track_number]` . (not sure if it's related though

**Reply by hundredfire** [reply]
Could it be because youtube music is still being synchronised in the background? Even though the playlist does appear in Music assistant's library? What I find strange is that I can play the playlist using the web interface without any issue though :-/

**Reply by OzGav** [reply]
You could use the mass.search service call and there you will find the library uri and you could use that

**Reply by hundredfire** [reply]
I may have messed up something during installation because I can't find the mass.search service. This could be the reason it's failing to find the relevant playlist? I'm off to troubleshoot why this service isn't there..

**Reply by OzGav** [reply, marked as answer]
Have you the latest version of the integration?

**Reply by hundredfire** [reply]
OK, I'll admit I'm a newbie in home assistant (installed it yesterday), but I will also say I'm a **huge** dumb-dumb. I installed music assistant server but completely flew over installing HACS and the custom integration. I'm sorry I wasted your time. I'm not deleting this so it may serve as a lesson to properly read documentation AND not to try new things late night and sleep-deprived.
Thank you again for your help. You've got a great project going, congratulations
